merlin-server/src/test/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintsDSLCompilationServiceTests.java

```diff
@@ -277,6 +278,31 @@ export default() => {
     );
   }
 
+  @Test
+  void testClampBy() {
+    checkSuccessfulCompilation(
+        """
+            export default() => {
+              return Real.Resource("state of charge").rate().equal(Real.Value(4.0)).shiftBy(1000, -200)
+            }
+        """,
+        new ViolationsOf(new ShiftBy(new Equal<>(new Rate(new RealResource("state of charge")), new RealValue(4.0)), Duration.of(1000, Duration.MICROSECOND), Duration.of(-200, Duration.MICROSECOND)))
```

Comment:
:large_orange_diamond: This is a loooooong line, I suggest we add some breaks/formatting for easier parsing.